app/services/workflow_link_service.rb

```diff
@@ -0,0 +1,22 @@
+class WorkflowLinkService
+  attr_accessor :workflow_id
+
+  def initialize(workflow_id)
+    self.workflow_id = workflow_id
+  end
+
+  def link(options)
```

Comment:
Feel it's more straightforward to use `tags` than `options`, as parameter's name. Same thing for other methods.